src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -95,23 +94,39 @@ private void Commit(PresentationItem item, Model model, char? commitChar, Cancel
                         IgnoreTrimWhiteSpace = EditOptions.DefaultMinimalChange.DifferenceOptions.IgnoreTrimWhiteSpace
                     });
 
-                    // edit subject buffer (not view) because text changes are in terms of current document.
-                    using (var textEdit = this.SubjectBuffer.CreateEdit(editOptions, reiteratedVersionNumber: null, editTag: null))
+                    var textChangesInCurrentViewSnapshot = new List<TextChange>();
+
+                    foreach (var triggerSnapshotTextChange in textChangesInTriggerSnapshot)
                     {
-                        for (int iChange = 0; iChange < textChanges.Length; iChange++)
+                        var currentSpanInBuffer = new SnapshotSpan(
+                            model.TriggerSnapshot,
+                            triggerSnapshotTextChange.Span.ToSpan());
+
+                        // Try mapping the item span against the trigger snapshot to the ViewBuffer's trigger snapshot.
+                        // Then map that forward to the ViewBuffer's current snapshot.
+                        var originalSpanInView = model.GetViewBufferSpan(triggerSnapshotTextChange.Span).TextSpan.ToSnapshotSpan(model.ViewTriggerSnapshot);
```

Comment:
why is the latter part of this expression necessary?  You alreasy have the viewbuffer span... but then you take it, unpack it, and then repack it.  I'm not really getting why.